### PR TITLE
[receivers/hostmetrics] add conntrack metrics

### PR DESCRIPTION
**Pull request by codeboten**
For linux only, adding metrics about connection tracking table.

### CODE REVIEW

File: receiver/hostmetricsreceiver/internal/scraper/networkscraper/metadata.yaml [hunk 1]
```diff
@@ -159,3 +159,22 @@ metrics:
...
+
+  system.network.conntrack.max:
+    enabled: true
+    description: The limit for entries in the conntrack table.
+    unit: "{entries}"
+    sum:
+      value_type: int
+      aggregation: cumulative
+      monotonic: false
+
```
Comment (mwear) [inline receiver/hostmetricsreceiver/internal/scraper/networkscraper/metadata.yaml:180]:
```suggestion

```

**Review by mwear — APPROVED** [review]
LGTM

**Comment by dmitryax** [comment]
Why do we want these metrics to be emitted by default? they seems like a good candidate for optional metrics

**Comment by dmitryax** [comment]
We don't have any strict guidelines to determine that, probably we should have it. Usually it's just up to code owner.

I usually follow the following reasoning: 
- The default set of metrics is enough to monitor a system at whole to determine its healthiness.
- Optional metrics are additional nice-to-have metrics that are not essential to determine system healthiness.

We also should take into consideration how stable a receiver is. If it has been used in production for some time, adding default metrics can affect downsteam backends that will unexpectedly receive additional load caused by added metrics with another collector upgrade.

**Comment by codeboten** [comment]
> Why do we want these metrics to be emitted by default? they seems like a good candidate for optional metrics

I agree, have disabled them by default. PTAL

**Comment by TylerHelmuth** [comment]
> We don't have any strict guidelines to determine that, probably we should have it. Usually it's just up to code owner.
> 
> I usually follow the following reasoning:
> 
> * The default set of metrics is enough to monitor a system at whole to determine its healthiness.
> * Optional metrics are additional nice-to-have metrics that are not essential to determine system healthiness.
> 
> We also should take into consideration how stable a receiver is. If it has been used in production for some time, adding default metrics can affect downsteam backends that will unexpectedly receive additional load caused by added metrics with another collector upgrade.

These thoughts would be awesome in the ["Adding metrics to existing receivers"](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#adding-metrics-to-existing-receivers) section of the contributing guide 👀

File: receiver/hostmetricsreceiver/internal/scraper/networkscraper/metadata.yaml [hunk 2]
```diff
@@ -159,3 +159,22 @@ metrics:
...
+  system.network.conntrack.max:
+    enabled: false
+    description: The limit for entries in the conntrack table.
+    unit: "{entries}"
+    sum:
+      value_type: int
+      aggregation: cumulative
+      monotonic: false
+
```
Comment (dmitryax) [inline receiver/hostmetricsreceiver/internal/scraper/networkscraper/metadata.yaml:180]:
```suggestion

```

**Review by dmitryax — APPROVED** [review]

File: receiver/hostmetricsreceiver/internal/scraper/networkscraper/metadata.yaml [hunk 3]
```diff
@@ -159,3 +159,21 @@ metrics:
...
+  system.network.conntrack.max:
+    enabled: false
+    description: The limit for entries in the conntrack table.
+    unit: "{entries}"
+    sum:
+      value_type: int
+      aggregation: cumulative
+      monotonic: false
```
Comment (dmitryax) [inline receiver/hostmetricsreceiver/internal/scraper/networkscraper/metadata.yaml:179]:
```suggestion
      monotonic: false

```

**Review by dmitryax — APPROVED** [review]

**Pull request merged**